test/io/test_io.cpp

```diff
@@ -1032,14 +1043,180 @@ TYPED_TEST (PLYPointCloudTest, LoadPLYFileColoredASCIIIntoPointCloud)
   // EXPECT_TRUE (cloud_rgb.is_dense); // this is failing and it shouldnt?
 
   // scope cloud data
-  ASSERT_EQ (cloud_rgb[0].rgba, PLYTest::rgba_1_);
-  ASSERT_EQ (cloud_rgb[1].rgba, PLYTest::rgba_2_);
-  ASSERT_EQ (cloud_rgb[2].rgba, PLYTest::rgba_3_);
-  ASSERT_EQ (cloud_rgb[3].rgba, PLYTest::rgba_4_);
+  ASSERT_EQ (cloud_rgb[0].rgba, FileTest::rgba_1_);
+  ASSERT_EQ (cloud_rgb[1].rgba, FileTest::rgba_2_);
+  ASSERT_EQ (cloud_rgb[2].rgba, FileTest::rgba_3_);
+  ASSERT_EQ (cloud_rgb[3].rgba, FileTest::rgba_4_);
 }
 
 //////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+class OBJTest : public FileTest
+{
+  public:
+
+  OBJTest ()
+  {
+    path_ = "obj_color_mesh.obj";
+
+    std::ofstream fs;
+    fs.open (path_.c_str ());
+    fs  << "vn 1.000000 0.000000 0.000000\n"
+        << "v 1.000000 0.000000 0.000000 0.678431 0.509804 0.600000\n"
+        << "vn 0.000000 1.000000 0.000000\n"
+        << "v 0.000000 1.000000 0.000000 0.905882 0.149020 0.654902\n"
+        << "vn 0.000000 0.000000 1.000000\n"
+        << "v 0.000000 0.000000 1.000000 1.000000 0.019608 0.792157\n"
+        << "vn -1.000000 0.000000 0.000000\n"
+        << "v -1.000000 0.000000 0.000000 0.188235 0.117647 0.737255\n"
+        << "vn 0.000000 -1.000000 0.000000\n"
+        << "v 0.000000 -1.000000 0.000000 0.298039 0.243137 0.776471\n"
+        << "vn 0.000000 0.000000 -1.000000\n"
+        << "v 0.000000 0.000000 -1.000000 0.019608 0.517647 0.439216\n"
+        << "f 1//1 2//2 3//3\n"
+        << "f 1//1 3//3 5//5\n"
+        << "f 1//1 5//5 6//6\n"
+        << "f 1//1 6//6 2//2\n"
+        << "f 4//4 2//2 6//6\n"
+        << "f 4//4 6//6 5//5\n"
+        << "f 4//4 5//5 3//3\n"
+        << "f 4//4 3//3 2//2\n";
+    fs.close ();
+
+    // Test colors from first 4 vertices
+    rgba_1_ =  static_cast<uint32_t> (255) << 24 | static_cast<uint32_t> (173) << 16 |
+              static_cast<uint32_t> (130) << 8 | static_cast<uint32_t> (153);
+    rgba_2_ =  static_cast<uint32_t> (255) << 24 | static_cast<uint32_t> (231) << 16 |
+              static_cast<uint32_t> (38) << 8 | static_cast<uint32_t> (167);
+    rgba_3_ =  static_cast<uint32_t> (255) << 24 | static_cast<uint32_t> (255) << 16 |
+              static_cast<uint32_t> (5) << 8 | static_cast<uint32_t> (202);
+    rgba_4_ =  static_cast<uint32_t> (255) << 24 | static_cast<uint32_t> (48) << 16 |
+              static_cast<uint32_t> (30) << 8 | static_cast<uint32_t> (188);
+  }
+};
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+TEST_F (OBJTest, LoadOBJFileColoredASCIIIntoBlob)
+{
+  int res;
+  uint32_t rgba;
+
+  PCLPointCloud2 cloud_blob;
+  uint32_t ps;
+  int32_t offset = -1;
+
+  // check if loading is ok
+  res = loadOBJFile (path_, cloud_blob);
+  ASSERT_EQ (res, 0);
```

Comment:
True. I'll make the changes nevertheless. Better stick to good practices all the way.